consensus/src/threshold_simplex/types.rs

```diff
@@ -1392,16 +1434,26 @@ impl<V: Variant, D: Digest> Read for Finalize<V, D> {
     fn read_cfg(reader: &mut impl Buf, _: &()) -> Result<Self, Error> {
         let proposal = Proposal::read(reader)?;
         let proposal_signature = PartialSignature::<V>::read(reader)?;
+        let seed_signature = PartialSignature::<V>::read(reader)?;
+        if proposal_signature.index != seed_signature.index {
+            return Err(Error::Invalid(
+                "consensus::threshold_simplex::Finalize",
+                "mismatched signatures",
+            ));
+        }
         Ok(Finalize {
             proposal,
             proposal_signature,
+            seed_signature,
         })
     }
 }
 
 impl<V: Variant, D: Digest> EncodeSize for Finalize<V, D> {
```

Comment:
In a future PR, I suspect we could further abstract `Notarize/Finalize` now that they are ~identical (other than namespace)